doc/nrf/ug_partition_manager.rst

```diff
@@ -0,0 +1,203 @@
+
+   #ifndef CONFIG_SOC_NRF9160
+
+   # 'provision' is the name of the placeholder partition.
+   provision:
+     # This partition is stored at the very end of flash.
+     placement: last
+
+   #endif /* CONFIG_SOC_NRF9160 */
+
+.. _pm_yaml_partition_types:
+
+Partition types
+~~~~~~~~~~~~~~~
+
+It is required that each :file:`pm.yml` defines exactly one *image partition*.
+This is done by using the same name for the partition as the image name.
+
+All other partitions which have the `placement` property set are *placeholder
+partitions*.
+
+Partitions which have the `sub_partitions` property set are *phony partitions*,
+and do not occupy space in flash.
+
+In addition to the configuration provided in :file:`pm.yml`, each partition with the `placement` property set must have a corresponding size configuration in the image's ``Kconfig`` file.
+The format of this configuration is as follows:
+
+.. code-block:: none
+
+   config PM_PARTITION_SIZE_[PARTITION_NAME]
+   hex "Flash space reserved for [partition name]."
+   default 0xD000
+   help
+     Flash space set aside for [partition name]. Note, the name
+     of this configuration needs to match the requirements set by the
+     script 'partition_manager.py'. See pm.yaml.
+
+.. _pm_build_system:
+
+Build system
+============
+This section describes how the Partition Manager is included by the Zephyr build system.
+
+If one or more sub-images are included in a build, a set of properties for that sub-image is appended to a global list.
+
+These properties are:
+
+Path to :file:`pm.yml`
+   * Build directory path
+   * Path to generated include folder
+
+Once CMake finishes configuring the sub-images, the Partition Manager script is executed in configure time (`execute_process`) with the aforementioned list as argument.
+The configurations generated by the Partition Manager script are imported as CMake variables. See :ref:`pm_generated_output_and_usage`.
+
+.. _pm_generated_output_and_usage:
+
+Generated output and usage
+==========================
+For each sub-image and the root app, Partition Manager generates two files, one C header file, and one Kconfig file.
+The C header file is used in the C code while the Kconfig file is imported in CMake.
+Both these files contain the start address and size of all partitions.
+The Kconfig file additionally contains the build directory and generated include folder for each image.
+
+C code usage
+   When Partition Manager is enabled, all source files are compiled with the define ``USE_PARTITION_MANAGER`` set to 1.
+   This allows the preprocessor to choose what code to include, depending on whether or not Partition Manager is being used.
+
+   .. code-block:: C
+
+      #if USE_PARTITION_MANAGER
+      #include <pm_config.h>
+      #define NON_SECURE_APP_ADDRESS PM_APP_ADDRESS
+      #else
+      ...
+
+
+CMake usage
+   The CMake variables from Partition Manager are typically used through generator expressions.
```

Comment:
What CMake variables exactly are provided? Is this something users can define? Or does Partition Manager define some of its own (always? depending on configuration? etc.)

Reply:
This behavior will change as all `PM_` properties will be added to a specific target. Will ensure that this is documented.